src/backend/src/oisy_user.rs

```diff
@@ -0,0 +1,35 @@
+use crate::{Candid, StoredPrincipal, VMem};
+use candid::Principal;
+use ic_stable_structures::StableBTreeMap;
+use shared::types::{
+    user_profile::{GetUsersRequest, OisyUser, StoredUserProfile},
+    Timestamp,
+};
+use std::ops::Bound;
+
+const DEFAULT_LIMIT_GET_USERS_RESPONSE: usize = 10_000;
+
+pub fn get_oisy_users(
+    request: &GetUsersRequest,
+    user_profile_map: &StableBTreeMap<(u64, StoredPrincipal), Candid<StoredUserProfile>, VMem>,
+) -> (Vec<OisyUser>, u64) {
+    let limit_users_size: usize = match request.matches_max_length {
```

Comment:
nth: maybe we can extract this in a function

Reply:
done